checks.d/elastic.py

```diff
@@ -368,6 +460,66 @@ def _process_stats_data(self, nodes_url, data, stats_metrics, config):
                     node_data, metric, *desc, tags=config.tags,
                     hostname=metric_hostname)
 
+    def _process_pshard_stats_data(self, data, config, pshard_stats_metrics):
+        for metric, desc in pshard_stats_metrics.iteritems():
+            self._process_metric(data, metric, *desc, tags=config.tags)
+
+    def _process_shard_level_data(self, data, config, shard_level_metrics):
+        for index_name, index_data in data['indices'].iteritems():
+            for i_str, pshard_and_replicas in index_data['shards'].iteritems():
+                # Do we have more than one replica for this shard ?
+                count_replicas = len(pshard_and_replicas) > 2
+                replica_number = 0
+
+                for shard in pshard_and_replicas:
+                    # Let's get the host tag
+                    node = shard['routing']['node']
+
+                    # Let's compute our shard name
+                    shard_role = "replica"
+                    if shard['routing']['primary']:
+                        shard_name = 'P' + i_str
+                        shard_role = "primary"
+                    elif count_replicas:
+                        replica_number += 1
+                        shard_name = 'R' + i_str + '_{0}'.format(replica_number)
```

Comment:
you could write `shard_name = 'R'{0}_{1}'.format(i_str, replica_number)` instead